nbbuild/antsrc/org/netbeans/nbbuild/extlibs/MavenSkeletonProject.java

```diff
@@ -66,13 +67,15 @@ void execute() throws BuildException {
                 modules.add("nbbuild");
             } else {
                 Path nbAllPath = nball.toPath();
-                modules = new TreeSet<>(
-                        Files.walk(nbAllPath)
-                                .filter(p -> Files.exists(p.resolve("external/binaries-list")))
-                                .map(p -> nbAllPath.relativize(p))
-                                .map(p -> p.toString())
-                                .collect(Collectors.toSet())
-                );
+                modules = new TreeSet<>();
+                try ( Stream<Path> walk = Files.walk(nbAllPath)) {
+                    modules = new TreeSet<>(
```

Comment:
the 2x new TreeSet() is because you don't want it to be null on exception? Wouldn't it be easier to initialize the private field as empty set so that you don't have to do this here twice?